### DISCUSSION TITLE
How to make the sensor last as long as possible (bypass kill switch ?)

### DISCUSSION THREAD

**Discussion by shodanx2** [NightscoutFoundation/xDrip#2720] [General]
Hello,

I have just had a freestyle glucose sensor installed.

I am not diabetic so I will not be getting another one.

I just want to know about my blood glucose and relate that to how I eat and how I feel.

So I don't really care about the absolute values only the relative values and I don't really care about the exact accuracy as I will not use it to take insulin. 

With that in mind, I would like to make my sensor last as long as it really can and not have its life cut short by a manufacturer kill switch.

So I installed the sensor and so far I have not synced with anything, I haven't installed the app and I have a special wiped blank android 9 phone with no internet access that I will dedicate to this experiment for only recording my blood glucose levels.

Is there something I should do because scanning it for the first time with xdrip ? I imagine the counter started when the sensor was taken out of its special container. Or is it the first time it is scanned with NFC ?

Please advise !

**Comment by steve8x8** [comment]
Did you check https://www.supersapiens.com/ for access to sensors (with the same lifetime limit though)?

**Reply by shodanx2** [reply]
I had not heard of that service, however the price is more than half my house mortgage, I cannot afford this

**Comment by PhilTU** [comment]
Is it a UK/EU Libre 1 or US 14 day sensor ?  The run time goes from when you activate by NFC scanning. xDrip+ will collect data for 14.5 days, half a day after the official timeout.  Some Libre 1's can be "extended" by iDroplet v5 - UK/EU only I suspect.

**Reply by shodanx2** [reply]
Here is what is on the box 

![image](https://user-images.githubusercontent.com/10621885/223553076-0c666e28-445b-4d99-9afa-d13b6b8e309f.png)

This is what the sensor looks like 

![image](https://user-images.githubusercontent.com/10621885/223553211-eb56cdea-3970-4cd0-b9da-7550df4cbb05.png)

I think the closest to a model number on the box is (01)00093815715344

I have side loaded this apk

https://github.com/NightscoutFoundation/xDrip/releases/download/2023.03.06/xDrip-plus-20230306-cb16679.apk

with this command

adb -s XXXXXXXXXX install xDrip-plus-20230306-cb16679.apk

Onto a fresh wiped phone, moto z3 running android 9 that has never been connected to the internet

I started the app and it asks for which device, I said "Libre"

Then it asked for device bridge, I don't have a bridge

I see  "Libre2 Patched" is an option, I imagine this is somehow reprogramming the sensor with a new firmware ? How can I do that, I am very good with electronics, I reprogrammed the ECU on my car when the orignal chip fried (dealer wanted 600$ plus would only do it to a blank, new 1200$ unit, I got reprogrammed one with my old data into a 20$ unit from the junkyard)

Although I suspect I was probably supposed to upload a hacked firmware into it before sticking it on my arm eh ?

**Reply by shodanx2** [reply]
I have a rasperry pi 4 1gb and a raspberry pi zero 2, could I make a device bridge with it ?

I also have a usb NFC reader on my PC and it also has bluetooth

**Reply by PhilTU** [reply]
OK that's a Libre 1 EU/UK style I think. From memory Libre Alarm is the right data source to scan this with NFC. https://xdrip.readthedocs.io/en/latest/install/libreNFC/

iDroplet app will attempt to extend the life of that sensor.

**Reply by shodanx2** [reply]
Thank you that worked

So here is what happened

This morning, I had tried xdrip and couldn't figure it out.
I tried figuring out a bridge device or the patched libre (what is patched mean, can I hack this thing's firmware , maybe recharge the battery and put  in a new needle ?)

I did not try the sony alarm thing, I don't have a sony watch !? (This should be made clearer, because I was actually so close to making it work, your instructions work !)

So, next I tried installing the librelink app on my wiped, never internet connect phone, I side-loaded the app
But it refused to start, said "you must install this via the google play store" 

So, I relented, I installed the librelink app on my good phone, probably accepted to give away my soul to the corporation 
They make you accept 3 contracts I think, filled all the information with fake data, fake email account, the works, still feels dirty
I bet they will sell my medical data to everyone and especially insurance companies and they'll find a way to screw me over when I'm old with that data !


Anyway, did that, 

It armed the sensor self-destruct sequence, probably patched the sensor against firmware attacks too

![image](https://user-images.githubusercontent.com/10621885/223694143-c5ceb8a1-d1d0-4840-bb77-66cc7fda67b1.png)

Started getting readings

![image](https://user-images.githubusercontent.com/10621885/223694240-a643fbd4-d7ca-4829-b866-519b68a6c63f.png)

Looks good

![image](https://user-images.githubusercontent.com/10621885/223694288-a1f8ebc2-c53c-41f0-9f6e-9da159ee2c9f.png)


After reading your instructions, I tried adding the libre alarm source with the sony watch


![image](https://user-images.githubusercontent.com/10621885/223693005-25f0f623-3132-492f-a2d2-f430a5d2f49d.png)


Turned on NFC in settings

I tried putting in my values but it refused "values out of range"

I ended up typing 50 and 60 I believe, I don't know what units it was expecting

Ah, now I see it was supposed to me mg/dl but with use mmol/L here in Canada

And it works now ! Absolute values are wrong but the relative values look similar

![image](https://user-images.githubusercontent.com/10621885/223695298-b9188845-3790-494e-8e5a-ead8e166b773.png)

I will take screenshots regularly on both devices and report back in this thread when the sensor expires



So after a while of it not working

**Reply by PhilTU** [reply]
Ok, the instructions are a little old as use of xDrip+ for scanning Libre 1 sensor is not common, SW3 watch was main use as it can scan itself..

The official app should install on your "special" phone too, from Play store. Installing apk should work but there are Geo checks so it is more complicated.

The "Patched app" is a version of Librelink that feeds data values into xDrip+, it is used with Libre 2 sensors (EU/UK only) that have Bluetooth transmission. There is no patching of the sensor. The only app that affects the "kill switch" is iDroplet which has a Facebook support group. It may not work with a Canadian sensor (I don't know).

https://xdrip.readthedocs.io/en/latest/ might help with xDrip you can switch it to mmol 1 mmol/l = 18 mg/dl

**Comment by jlecount** [comment]
I'm using dexcom g6 sensors and get 10d but then can usually restart it for another 10d before they actually start getting unreliable.

**Reply by shodanx2** [reply]
Thanks, what does the restart procedure look like ? 
I'm fine with bad absolute values, I just want relative values

Also, what is the sensing element, it looks like just a needle ? 
Is it just a galvanic reaction on a platinum coated needle ? 
I couldn't observe the sensor before deploying it, is the needle hollow, does it draw blood into the device ?
